src/network/taraxa_capability.cpp

```diff
@@ -981,10 +981,10 @@ void TaraxaCapability::onNewTransactions(std::vector<taraxa::bytes> const &trans
     for (auto &it : transactionsToSend) {
       sendTransactions(it.first, it.second);
     }
-    boost::unique_lock<boost::shared_mutex> lock(peers_mutex_);
+    boost::shared_lock<boost::shared_mutex> lock(peers_mutex_);
     for (auto &it : transactionsHashToSend) {
       for (auto &it2 : it.second) {
-        peers_[it.first]->markTransactionAsKnown(it2);
+        if (peers_[it.first]) peers_[it.first]->markTransactionAsKnown(it2);
```

Comment:
same mistake as before.... operator [] on map creates new element in case requested one does not exist. Use find instead https://en.cppreference.com/w/cpp/container/unordered_map/find

Reply:
thanks